### PR TITLE
Adding linting to ci

### PR DESCRIPTION
**Pull request by IanWoodard**
Building off of #60, adding prettier and eslint to the ci so we can enforce the formatting.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,10 +3,16 @@
   "description": "Jsonnet libraries to help with GoCD",
   "type": "module",
   "scripts": {
-    "test": "ava"
+    "test": "ava",
+    "eslint": "eslint --config './.eslintrc.json' .",
```
Comment (bukzor-sentryio) [inline package.json:7]: That's the default value for --config, so I'm not sure this should be here.

https://eslint.org/docs/latest/use/configure/configuration-files#configuration-file-formats

P4: Not a blocking issue, at all.

File: package.json [hunk 2]
```diff
@@ -3,10 +3,16 @@
...
+    "prettier": "prettier .",
+    "lint": "npm run eslint && npm run prettier -- --check",
+    "lint:fix": "npm run eslint -- --fix && npm run prettier -- -w"
   },
   "devDependencies": {
-    "ava": "^5.3.0"
+    "ava": "^5.3.0",
+    "eslint": "^8.53.0",
+    "prettier": "^3.1.0"
```
Comment (bukzor-sentryio) [inline package.json:15]: I want prettier to continue working even when this repo stops using js (we'll convert to python someday soon-ish). Please use pre-commit for these responsibilities. For example:

https://github.com/getsentry/sentry/blob/07f4f07b9/.pre-commit-config.yaml#L114

P2: I'd rather not have to revisit this in future, but you can punt if you decide it's too annoying.

**Review by bukzor-sentryio — CHANGES_REQUESTED** [review]
Please configure lint & fix via pre-commit?
But I'll still approve if you decide it's not worthwhile at the moment.

File: package.json (hunk 2 above)
Comment (IanWoodard) [reply package.json:15]: punting

**Pull request merged**